platforms/jvm/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/junit/JUnitTestFramework.java

```diff
@@ -58,14 +60,16 @@ public class JUnitTestFramework implements TestFramework {
     private final boolean useImplementationDependencies;
     private final Factory<File> testTaskTemporaryDir;
     private final Provider<Boolean> dryRun;
+    private final ObjectFactory objects;
 
     public JUnitTestFramework(Test testTask, DefaultTestFilter filter, boolean useImplementationDependencies) {
-        this(filter, useImplementationDependencies, new JUnitOptions(), testTask.getTemporaryDirFactory(), testTask.getDryRun());
+        this(filter, useImplementationDependencies, testTask.getProject().getObjects().newInstance(JUnitOptions.class), testTask.getTemporaryDirFactory(), testTask.getDryRun(), testTask.getProject().getObjects());
```

Comment:
🤔 There doesn't seem a need for the private constructor here?  You could just collapse these into a single constructor and avoid unpacking the contents of `testTask` to supply args to the private one.

Reply:
We are using private constructor in `copyWithFilters` method few lines below.

I think that is the only reason we have it.

Reply:
OK, I missed that.